### PR TITLE
minor changes

### PR DESCRIPTION
**Pull request by yuw006**
item1
item2
item3
item6

item7

https://user-images.githubusercontent.com/63177486/139024306-ce6672fd-c733-4817-9cd7-7a5d0dec5ea9.mp4

### CODE REVIEW

**Review by thai-truong — APPROVED** [review]
Lgtm, but wait for Kyle's approval before you merge

**Review by kyle1373 — APPROVED** [review]
ltgm! Nice changes

**Pull request merged**
